src/pages/events/createEvents/CreateEvent.tsx

```diff
@@ -0,0 +1,473 @@
+    header: 'Evaluator 1',
+    name: 'Name / Surname',
+    nameValue: '',
+    position: 'Position',
+    positionValue: '',
+    email: 'Email',
+    emailValue: '',
+  })
+
+  const [evaluateeObjInfo] = useState<IEvaluatee>({
+    id: uuidv4(),
+    header: '',
+    name: 'Name / Surname',
+    nameValue: '',
+    position: 'Position',
+    positionValue: '',
+    date: 'Hire date',
+    dateValue: null,
+    salary: 'Monthly Salary',
+    salaryValue: null,
+  })
+
+  const navigateToCriteriasAndSendDataToContext = () => {
+    dispatch({type: createEventReducerTypes.eventTitle, eventTitle})
+    dispatch({type: createEventReducerTypes.evaluators, evaluators: evaluatorsList})
+    dispatch({type: createEventReducerTypes.evaluatees, evaluatees: evaluateesList})
+    navigate('criteria')
+  }
+
+  const label = {inputProps: {'aria-label': 'Checkbox demo'}}
+
+  const handleChangeCheckbox = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setIsTopManagerChecked(e.target.checked)
+  }
+
+  const onChangeEvaluatorName = (id: string, value: string) => {
+    const newArray = evaluatorsList.map((item: IEvaluator) => {
+      if (item.id === id) {
+        return {...item, nameValue: value}
+      }
+      return item
+    })
+    setEvaluatorsList(newArray)
+  }
+
+  const onChangeEvaluatorPosition = (id: string, value: string) => {
+    const newArray = evaluatorsList.map((item: IEvaluator) => {
+      if (item.id === id) {
+        return {...item, positionValue: value}
+      }
+      return item
+    })
+    setEvaluatorsList(newArray)
+  }
+
+  const onChangeEvaluatorEmail = (id: string, value: string) => {
+    const newArray = evaluatorsList.map((item: IEvaluator) => {
+      if (item.id === id) {
+        return {...item, emailValue: value}
+      }
+      return item
+    })
+    setEvaluatorsList(newArray)
+  }
+
+  const onChangeEvaluateeName = (id: string, value: string) => {
+    const newArray = evaluateesList.map((item: IEvaluatee) => {
+      if (item.id === id) {
+        return {...item, nameValue: value}
+      }
+      return item
+    })
+    setEvaluateesList(newArray)
+  }
+
+  const onChangeEvaluateePosition = (id: string, value: string) => {
+    const newArray = evaluateesList.map((item: IEvaluatee) => {
+      if (item.id === id) {
+        return {...item, positionValue: value}
+      }
+      return item
+    })
+    setEvaluateesList(newArray)
+  }
+
+  const onChangeEvaluateeDate = (id: string, value: string) => {
+    const newArray = evaluateesList.map((item: IEvaluatee) => {
+      if (item.id === id) {
+        return {...item, dateValue: value}
+      }
+      return item
+    })
+    setEvaluateesList(newArray)
+  }
+
+  const onChangeEvaluateeSalary = (id: string, value: string) => {
+    const newArray = evaluateesList.map((item: IEvaluatee) => {
+      if (item.id === id) {
+        return {...item, salaryValue: value}
+      }
+      return item
+    })
+    setEvaluateesList(newArray)
+  }
```

Comment:
You can use only 1 function for all of this, it's just taking place

Reply:
done